### ISSUE TITLE
Drag-and-Drop Support in Reactive Scheduler

### ISSUE THREAD

**Issue by SergeyAlexeev** [DevExpress/devextreme-reactive#1905]
# The Problem

Users cannot edit an appointment's start/end date and time via drag-and-drop. 

# The Proposed Solution

We plan to add a new `DragDropProvider` plugin that allows users to drag-and-drop appointments. It will contain properties that allow you to customize the appearance that an appointment has while a user drags it and prevent particular appointments from being dragged.

The following example shows this feature's API:

```jsx
<Scheduler>
  {/* ... */}

  <EditingState
    onCommitChanges={/* Save changes here... */}
  />
  <DragDropProvider
    draftAppointmentComponent={/* Customize the appointment being dragged */}
    sourceAppointmentComponent={/* Customize a copy of the appointment being dragged displayed in the appointment's previous location */}
    containerComponent={/* Customize the appointment's container */}
    allowDrag={/* Prevent a particular appointment from being dragged */}
  />
</Scheduler>
```

# We Need Your Feedback

## Take a Quick Poll

[Do you find the new drag-and-drop functionality in Reactive Scheduler useful?](https://docs.google.com/forms/d/e/1FAIpQLSfWRG6N5WGoyKUQ39UHau2yI-T3wDNmXS4EnH7YX35Vj5T5Jg/viewform?usp=sf_link)

## Live Sandbox

You can check out the [prototype](https://codesandbox.io/s/9y3y3w9q1w).

A package that contains the described functionality has the `next` tag:

```
npm i --save @devexpress/dx-react-core@next @devexpress/dx-react-scheduler@next
npm i --save @devexpress/dx-react-scheduler-material-ui@next
```

[Documentation](https://devexpress.github.io/devextreme-reactive/@next/react/scheduler/docs/guides/editing#drag-and-drop-editing).

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by MaximKudriavtsev** [comment]
We have been released the `@next` version of [DevExtreme Reactive Scheduler](https://devexpress.github.io/devextreme-reactive/@next/react/scheduler/docs/guides/getting-started/) with this feature. Feel free to provide your feedback about its functionality and [documentation](https://devexpress.github.io/devextreme-reactive/@next/react/scheduler/docs/guides/editing/#drag-and-drop-editing).

A package that contains the described functionality has the `next` tag:

```
npm i --save @devexpress/dx-react-core@next @devexpress/dx-react-scheduler@next
npm i --save @devexpress/dx-react-scheduler-material-ui@next
```

**Comment by MaximKudriavtsev** [comment]
We have released the `1.11.0` version of [DevExtreme Reactive Scheduler](https://devexpress.github.io/devextreme-reactive/react/scheduler/docs/guides/getting-started/) with this feature.

A package that contains the described functionality has the `latest` tag:

```
npm i --save @devexpress/dx-react-core @devexpress/dx-react-scheduler
npm i --save @devexpress/dx-react-scheduler-material-ui
```
